addons/website_slides/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -7,413 +7,585 @@ odoo.define('website_slides.fullscreen', function (require) {
+                    this.goPrevious();
+                    break;
+                case "ArrowRight":
+                    this.goNext();
+                    break;
+            }
+        },
+    });
+
+
+    /**
+     * This widget's purpose is to show content of a course, naviguating through contents
+     * and correclty display it. It also handle slide completion, course progress, ...
+     *
+     * This widget is rendered sever side, and attached to the existing DOM.
+     */
+    var Fullscreen = Widget.extend({
+        events: {
+            "click .o_wslides_fs_toggle_sidebar": '_onClickToggleSidebar',
+        },
+        custom_events: {
+            'change_slide': '_onChangeSlideRequest',
+            'toggle_sidebar': '_onToggleSidebar',
+            'slide_completed': '_onSlideCompleted',
+            'quiz_completed': '_onSlideCompleted',
+            'slide_go_next': '_onSlideGoToNext',
+        },
+        /**
+        * @override
+        * @param {Object} el
+        * @param {Object} slides Contains the list of all slides of the course
+        * @param {Int} defaultSlideId Contains the ID of the slide requested by the user
+        */
+        init: function (el, slides, defaultSlideId){
+            var result = this._super.apply(this,arguments);
+            this.initialSlideID = defaultSlideId;
+            this.slides = this._preprocessSlideData(slides);
+
+            var slide;
+            if (defaultSlideId) {
+                slide = this._findSlide({id: defaultSlideId});
+            } else {
+                slide = this.slides[0];
+            }
+
+            this.set('slide', slide);
+
+            // extract data for sidebar
+            var sidebarData = _.map(this.slides, function(slide) {
+                return {
+                    id: slide.id,
+                    hasNext: slide.hasNext,
+                    isQuiz: slide.isQuiz,
                 }
             });
+            this.sidebar = new Sidebar(this, sidebarData, {
+                id: slide.id,
+                isQuiz: slide.isQuiz,
+                hasNext: slide.hasNext,
+            });
+            return result;
+        },
+        /**
+         * @override
+         */
+        start: function (){
+            var self = this;
+            this.on('change:slide', this, this._onChangeSlide);
+            return this._super.apply(this, arguments).then(function () {
+                self._onChangeSlide(); // trigger manually once DOM ready, since slide content is not rendered server side
+            });
         },
         /**
+         * Extended to attach sub widget to sub DOM. This might be experimental but
+         * seems working fine.
+         *
+         * @override
+         */
+        attachTo: function (){
+            var defs = [this._super.apply(this, arguments)];
+            defs.push(this.sidebar.attachTo(this.$('.o_wslides_fs_sidebar')));
+            return $.when.apply($, defs);
```

Comment:
Should probably be `Promise.all(this._super.apply(this, arguments), this.sidebar.attachTo(this.$('.o_wslides_fs_sidebar')))`, to check.